Index: javaee/src/main/java/org/javasimon/javaee/SimonEjbInterceptor.java
===================================================================
--- javaee/src/main/java/org/javasimon/javaee/SimonEjbInterceptor.java	(revision 390)
+++ javaee/src/main/java/org/javasimon/javaee/SimonEjbInterceptor.java	(working copy)
@@ -18,7 +18,10 @@
 	 * Default prefix for EJB interceptor Simons if no "prefix" init parameter is used.
 	 */
 	public static final String DEFAULT_EJB_INTERCEPTOR_PREFIX = "org.javasimon.ejb";
-
+        /**
+         * Simon name prefix
+         */
+        protected String prefix = DEFAULT_EJB_INTERCEPTOR_PREFIX;
 	/**
 	 * Returns Simon name for the specified Invocation context.
 	 * By default it contains the prefix + method name.
@@ -29,7 +32,9 @@
 	 * @since 3.1
 	 */
 	protected String getSimonName(InvocationContext context) {
-		return DEFAULT_EJB_INTERCEPTOR_PREFIX + Manager.HIERARCHY_DELIMITER + context.getMethod().getName();
+		String className=context.getMethod().getDeclaringClass().getSimpleName();  
+		String methodName=context.getMethod().getName();  
+		return prefix + Manager.HIERARCHY_DELIMITER + className + Manager.HIERARCHY_DELIMITER + methodName;
 	}
 
 	/**
